packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -228,7 +228,7 @@ export default function TypographyPanel( {
 			immutableSet( value, [ 'typography', 'lineHeight' ], newValue )
 		);
 	};
-	const hasLineHeight = () => !! value?.typography?.lineHeight;
+	const hasLineHeight = () => value?.typography?.lineHeight !== undefined;
```

Comment:
I've kept this tweak to the line height value detection so it can correctly check any old numeric line-height values that might be `0` in existing blocks/styles.